### DISCUSSION TITLE
Detailed use of masks

### DISCUSSION THREAD

**Discussion by neriniccolodei** [OpenStitching/stitching#175] [General]
Hello, 

I have seen the implementation of masks in the stitching tutorial. I'd like to know if you have the code for using them in a more detailed way (e.g., not just `panorama = stitcher.stitch(images_to_stitch, masks_of_those_images)`). 

To clarify my need: 
1) I have 2 unprocessed frames.
2) I have processed these frames to achieve better feature extraction (filtering them)
3) Therefore, I use the processed frames to find features across the 2 original frames, and I specify areas where to look for features using masks.
4) Using `panorama = stitcher.stitch(processed_frames, masks)`, I get the panorama of the processed images.
5) I'd like to use the same transformation that was deducted here and apply it to original images to get the original frames panorama.

This is easy to do, when masks are not necessary, in the "detailed" code (e.g., the one in the stitching tutorial): right before stitching, one overwrites the variables `low_imgs`, `medium_imgs` and `final_imgs` loading the original ones. However, I am not able to make this when masks are at place, as I can only use them in the "simple" code case. 

Thank you and best regards

**Comment by lukasalexanderweber** [comment]
You could extend in https://github.com/OpenStitching/stitching/blob/main/stitching/stitcher.py

 stitch(self, images, feature_masks=[])

with

stitch(self, processed_images, original_images, feature_masks)

Where you use the processed images and the masks till you have the caneras and afterwards use the original images for the rest of the stitching

**Reply by neriniccolodei** [reply]
Thank you for your reply, I will try to implement as suggested. 
Best regards
